File: bwu/string/28-implement-strstr.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        m, n = len(haystack), len(needle)
        
        if not n:
            return 0
            
        if m < n:
            return -1
            
        for i in xrange(m - n + 1):
            for j in xrange(n):
                if haystack[i + j] != needle[j]:
                    break
                if j == n - 1:
                    return i
        return -1

File: captainhcg/go/28-implement-strstr.go
func strStr(haystack string, needle string) int {
	if needle == ""{
		return 0
	}
	end := len(haystack) - len(needle) + 1
	start := 0
	for start < end {
		offset := 0
		for offset < len(needle) {
			if start+offset < len(haystack) && needle[offset] == haystack[start+offset] {
				offset += 1
			} else {
				break
			}
			if offset == len(needle) {
				return start
			}
		}
		start += 1
	}
	return -1
}
